cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -47,16 +49,18 @@
      */
     public KafkaConnectDockerfile(String fromImage, Build connectBuild) {
         StringWriter stringWriter = new StringWriter();
+        StringWriter prefixStringWriter = new StringWriter();
+        PrintWriter prefixWriter = new PrintWriter(prefixStringWriter);
```

Comment:
Why is this needed? I don't understand this

Reply:
From [this article](https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds) that the parts have to be done in a specific order. I did not try the unordered parts. Do you think it works the same way?

Reply:
Ah, ok now I understand it better ... but:
* The naming is unintuitive => I do not think anyone would understand that the prefix means the initial stage here.
* I wonder if we want to have the _one stage per artifact_ approach there. Wouldn't it be better to have a single writer with something like this and generate only the build stage + the regular container instead of having many stages?
  ```
        printHeader(writer); // Print initial comment
        connectorPluginsPreStage(writer, connectBuild.getPlugins());
        from(writer, fromImage); // Create FROM statement
        user(writer, ROOT_USER); // Switch to root user to be able to add plugins
        proxy(writer); // Configures proxy environment variables
        connectorPlugins(writer, connectBuild.getPlugins());
  ```

I also think the tool linked by Tom would be worth investigating. Keeping it inside the operator and generating just a regular Dockerfile might make it easier to test it (given the different build on Kube and OCP, the multistage build needs to be tested there twice).